ERCS/erc-7828.md

```diff
@@ -0,0 +1,236 @@
+
+In case an address is not registered on L1, but only on a L2, the resolution can be processed using [ERC-3668](./eip-3668.md) and [ERC-2544](./eip-2544.md)'s  Wildcard Resolution.
+
+In the previous example `user@rollup.eth`, `user` would not be registered on L1.
+In this case the wallet can resolve `rollup.eth` to get a chain identifier as before and when attempting to resolve `user.rollup.eth` to get an address, it would fail and be redirected to the L2 gateway. Any answer from the gateway needs to be verified as explained in the EIP.
+
+#### Note: avoiding the http gateway
+
+In order to avoid contacting an external http gateway, we could define the gateway to be a ENS contract on the L2. In this way a wallet operator would need to only rely on a node following the L1 and a node following the L2.
+
+### L1 resolution
+
+Ethereum Mainnet and its testnets can be resolved to their corresponding chain identifiers using a centralized list, which remains unchanged from how it works today. Other L1 registrations are out of scope for this EIP.
+
+Mapping:
+```
+L1-TLD -> {L1_chain_id : chain_id; L1_ens_address : address;}
+```
+Example:
+```
+eth     -> {L1_chain_id : 0x1;      L1_ens_address : <ENS-address-on-mainnet>}
+sepolia -> {L1_chain_id : 0xaa36a7; L1_ens_address : <ENS-address-on-sepolia>}
+```
+
+### Note: clashes of L1 and TLD
+
+In the above proposal the ENS TLD and chain name coincide which may be confusing or incorrect in some cases. A more explicit approach could be to have an additional suffix for the chain name.
+Example:
+```
+name.eth.mainnet -> {L1_chain_id : 0x1;      L1_ens_address : <ENS-address-on-mainnet>}
+name.eth.sepolia -> {L1_chain_id : 0xaa36a7; L1_ens_address : <ENS-address-on-sepolia>}
+```
+### Note: default fallbacks
+
+If a user receives a legacy address without chain name, the wallet can:
+Refuse the address (safest)
+Default to Mainnet (unambiguous)
+Dynamically default to same chainID as sender (ambiguous and context-dependent but probably compatible with current use-cases)
+
+### Advanced patterns
+
+The `@` syntax described in this EIP is a restricted resolution case for ENS names whose main purpose it to be user-friendly. We can however support more advanced pattern in ENS.
+
+### Supported today
+
+For example a user might configure its name with multiple address such as
+```
+rollup1.user.eth -> {address : address; chain_id : chain_id} 
+rollup2.user.eth -> {address : address; chain_id : chain_id} 
+```
+Given `user.eth` as recipient a wallet could prompt the user to select a destination chain.
+Otherwise the user can be more explicit and give as recipient `rollup1.user.eth`.
+
+### Note: Speculative
+
+Alternatively we could store multiple addressed under the same domain as
+```
+user.eth -> { rollup1 : address * chain_id ; 
+              rollup2 : address * chain_id } 
+```
+if a syntax to access ENS records could be standardized, the user could be asked to be paid at
+`user.eth/rollup1`
```

Comment:
[RFC 8615]( https://en.wikipedia.org/wiki/Well-known_URI) defines something similar to this, an HTTP server would serve from `user.eth/.well-known/addresses.json` (we can come up with an exact filename). I suppose it depends on if we want this legible onchain